### ISSUE TITLE
Redesign layouts

### ISSUE THREAD

**Issue by pngwn** [pngwn/MDsveX#584]
The conversation around layouts has changed somewhat since the release of Sveltekit.

Questions:

- Do we need layouts anymore? Can we achieve the same functionality with kit's layouts? What about other tools and integrations?
- If we keep layouts can we somehow integrate directly with kit's layouts? Maybe we can use kits layouts for kit users but have builtin layouts as a fallback?
- If we get rid of them what do we do about custom components

Related issues:

- [ ] #556 
- [ ] #485 
- [ ] #423 
- [ ] #313 
- [ ] #312 
- [ ] #311 
- [ ] #290 
- [ ] #277 
- [ ] #230 
- [ ] #73 
- [ ] #303

**Comment by zmre** [comment]
There isn't an issue for it, but I don't think you can currently use data from, for example, a `+layout.ts` (so no server-side data available) and you also can't use stores like the page store, which tells you the current path of the current URL, for example, which makes it hard to put navigation elements into a "layout" for a mdsvex markdown file. I expect at least one if not both of these problems would be resolved if the kit layout system were being used and that would greatly simplify my code and keep me from having to do a bunch of data loading and displaying in the browser after first render to handle navigation stuff.

**Comment by pngwn** [comment]
Yes. I'm trying to figure how best to approach this issue both for kit and non-kit users. 

I'm partly waiting to see what impact svelte 5 will have on kit before I make any concrete decisions.

**Comment by Magnus-Ignitic** [comment]
Just created issue https://github.com/pngwn/MDsveX/issues/687. Not sure if related.
